packages/oauth-helper/src/index.ts

```diff
@@ -0,0 +1,468 @@
+  storeInstallation: (installation: Installation) => Promise<void>;
+  fetchInstallation: (query: InstallationQuery) => Promise<Installation>;
+}
+
+interface DevDatabase {
+  [key: string]: Installation;
+}
+const devDB: DevDatabase = {};
+
+// Default Install Store. Should only be used for development
+class MemoryInstallStore implements InstallationStore {
+
+  public storeInstallation(installation: Installation): Promise<void> {
+    console.log('Storing Access Token. Please use a real Installation Store for production!');
+    // db write
+    devDB[installation.team.id] = installation;
+    return new Promise<void>((resolve) => {
+      resolve();
+    });
+  }
+
+  public fetchInstallation(query: InstallationQuery): Promise<Installation> {
+    console.log('Retrieving Access Token from DB. Please use a real Installation Store for production!');
+    // db read
+    const item = devDB[query.teamId];
+    return new Promise<Installation>((resolve) => {
+      resolve(item);
+    });
+  }
+}
+
+// Needs to have all the data from OAuthV2Access result and OAuthAccess
+// result. This is a normalized shape.
+interface Installation {
+  team: {
+    id: string;
+    name: string;
+  };
+  enterprise?: {
+    id: string;
+    name: string;
+  };
+  bot?: {
+    token: string;
+    scopes: string[];
+    id?: string;
+    userId: string;
+  };
+  user: {
+    token?: string;
+    scopes?: string[];
+    id: string;
+  };
+  incomingWebhook?: {
+    url: string;
+    channel: string;
+    configurationUrl: string;
+  };
+  appId: string | undefined;
+  tokenType?: string;
+}
+
+// This is intentionally structurally identical to AuthorizeSourceData
+// from App. It is redefined so that this class remains loosely coupled to
+// the rest of Bolt.
+interface InstallationQuery {
+  teamId: string;
+  enterpriseId?: string;
+  userId?: string;
+  conversationId?: string;
+}
+
+// This is intentionally structurally identical to AuthorizeResult from App
+// It is redefined so that this class remains loosely coupled to the rest
+// of Bolt.
+interface AuthorizeResult {
+  botToken?: string;
+  userToken?: string;
+  botId?: string;
+  botUserId?: string;
+}
+
+// Default function to call when OAuth flow is successful
+function callbackSuccess(
+  installation: Installation,
+  metadata: string | undefined,
+  req: IncomingMessage,
+  res: ServerResponse,
+): void {
+  // TODO: We never do anything with metadata. Should we?
+  // TODO: this if statement doesn't match the proposal one
+  // if (!classicAuth || scopes.includes('bot')
+  // How do i check if something is classicAuth? and bot scope doesn't make sense here
+  // TODO: Redirect only works for v2 OAuth as v1 is missing appId
+  if (installation.tokenType === 'bot') {
+    // redirect back to slack
+    // TODO: this redirects back to workplace general instead of appHome tab
+    const redirectUrl = `https://app.slack.com/client/${installation.team.id}/${installation.appId}/home`;
```

Comment:
this redirects back to workplace general channel for me instead of appHome tab. I assume that is expected but wanted to double check

Reply:
I think we should provide some options for extension. Also, it's the same for metadata handlers.

Reply:
Can you elaborate on what you mean options for extensions? You want to all devs to pass some options to modify the default callbackSuccess function? I think in our current approach, the dev would have to provide their own success callback to run instead.

Reply:
>I think in our current approach, the dev would have to provide their own success callback to run instead.

That makes sense. I think it's fine too. 

> You want to all devs to pass some options to modify the default callbackSuccess function?

You're right. I was assuming that we're going to provide the default way to handle common patterns and may allow developers to customize some parts of it. But it's more like a class-based object-oriented approach. I don't have any strong opinion here.